src/openrct2/actions/MazeSetTrackAction.hpp

```diff
@@ -315,11 +308,10 @@ DEFINE_GAME_ACTION(MazeSetTrackAction, GAME_COMMAND_SET_MAZE_TRACK, GameActionRe
                         tileElement->AsTrack()->MazeEntryAdd(1 << segmentBit);
 
                         uint32_t direction1 = byte_993D0C[segmentBit];
-                        uint16_t nextElementX = floor2(previousSegmentX, 32) + CoordsDirectionDelta[direction1].x;
-                        uint16_t nextElementY = floor2(previousSegmentY, 32) + CoordsDirectionDelta[direction1].y;
+                        auto nextElement = previousSegment.ToTileStart() + CoordsDirectionDelta[direction1];
```

Comment:
as above. Try to get location into this name

Reply:
`nextElementLoc`